### DISCUSSION TITLE
How to override AutoLink node

### DISCUSSION THREAD

**Discussion by skopz356** [facebook/lexical#6638] [General]
Based on https://lexical.dev/docs/concepts/transforms I would like to create custom Node for AutoLink that would modify exportDom of AutoLink. But when I implement it, it does not work. When I paste link into, nothing happens. I checked the source code of AutoLinkPlugin and I think that using current implementation it is not possible. 

[Codesandbox](https://codesandbox.io/p/sandbox/mention-cursor-safari-forked-rx56wy?layout=%257B%2522sidebarPanel%2522%253A%2522EXPLORER%2522%252C%2522rootPanelGroup%2522%253A%257B%2522direction%2522%253A%2522horizontal%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522id%2522%253A%2522ROOT_LAYOUT%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522UNKNOWN%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522cm0xtzd5c0006206el8z1cd31%2522%252C%2522sizes%2522%253A%255B100%255D%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522EDITOR%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522EDITOR%2522%252C%2522id%2522%253A%2522cm0xtzd5c0002206e86fggz7x%2522%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522direction%2522%253A%2522horizontal%2522%252C%2522id%2522%253A%2522SHELLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522SHELLS%2522%252C%2522id%2522%253A%2522cm0xtzd5c0003206etkdfz4dz%2522%257D%255D%257D%255D%257D%252C%257B%2522type%2522%253A%2522PANEL_GROUP%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522direction%2522%253A%2522vertical%2522%252C%2522id%2522%253A%2522DEVTOOLS%2522%252C%2522panels%2522%253A%255B%257B%2522type%2522%253A%2522PANEL%2522%252C%2522contentType%2522%253A%2522DEVTOOLS%2522%252C%2522id%2522%253A%2522cm0xtzd5c0005206e5ep3690q%2522%257D%255D%257D%255D%252C%2522sizes%2522%253A%255B50%252C50%255D%257D%252C%2522tabbedPanels%2522%253A%257B%2522cm0xtzd5c0002206e86fggz7x%2522%253A%257B%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0xtzd5c0001206ej0fiqtxr%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252Findex.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%257D%252C%257B%2522id%2522%253A%2522cm0xtzj3t0002206e8dqzi7q4%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522FILE%2522%252C%2522filepath%2522%253A%2522%252Fsrc%252FEditor.tsx%2522%252C%2522state%2522%253A%2522IDLE%2522%252C%2522initialSelections%2522%253A%255B%257B%2522startLineNumber%2522%253A42%252C%2522startColumn%2522%253A1%252C%2522endLineNumber%2522%253A42%252C%2522endColumn%2522%253A1%257D%255D%257D%255D%252C%2522id%2522%253A%2522cm0xtzd5c0002206e86fggz7x%2522%252C%2522activeTabId%2522%253A%2522cm0xtzj3t0002206e8dqzi7q4%2522%257D%252C%2522cm0xtzd5c0005206e5ep3690q%2522%253A%257B%2522id%2522%253A%2522cm0xtzd5c0005206e5ep3690q%2522%252C%2522activeTabId%2522%253A%2522cm1668lrn002c206euym06cty%2522%252C%2522tabs%2522%253A%255B%257B%2522id%2522%253A%2522cm0xtzd5c0004206ewa4bniif%2522%252C%2522mode%2522%253A%2522permanent%2522%252C%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm0xtzm3o000y206e10kwqgus%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%252C%257B%2522type%2522%253A%2522UNASSIGNED_PORT%2522%252C%2522port%2522%253A0%252C%2522id%2522%253A%2522cm1668lrn002c206euym06cty%2522%252C%2522mode%2522%253A%2522permanent%2522%257D%255D%257D%252C%2522cm0xtzd5c0003206etkdfz4dz%2522%253A%257B%2522tabs%2522%253A%255B%255D%252C%2522id%2522%253A%2522cm0xtzd5c0003206etkdfz4dz%2522%257D%257D%252C%2522showDevtools%2522%253Atrue%252C%2522showShells%2522%253Afalse%252C%2522showSidebar%2522%253Atrue%252C%2522sidebarPanelSize%2522%253A15%257D)

**Comment by etrepum** [comment]
If you only want to change exportDOM, you can override that using the `html` property of the editor config (I don't think there's documentation for this property but you can read the types for it, it's very similar to `importDOM` and `exportDOM`).

You may need to specify `withKlass` in the node replacement otherwise some plugins may not work, although a few plugins will not work with any node replacements and if that is the case you could report a bug about that. https://lexical.dev/docs/concepts/node-replacement

**Reply by skopz356** [reply]
Okay, thank you for the response. Have a nice day!
